component/conprof/http/api.go

```diff
@@ -301,17 +301,28 @@ func queryAndDownload(c *gin.Context) error {
 	fn := func(pt meta.ProfileTarget, ts int64, data []byte) error {
 		fileName := fmt.Sprintf("%v_%v_%v_%v", pt.Kind, pt.Component, pt.Address, ts)
 		fileName = strings.ReplaceAll(fileName, ":", "_")
-		if param.DataFormat == meta.ProfileDataFormatSVG {
-			svg, err := ConvertToSVG(data)
-			if err == nil {
-				data = svg
-				fileName += ".svg"
-			}
-		}
 		if pt.Kind == meta.ProfileKindGoroutine {
 			fileName += ".txt"
+		} else {
+			isSVG := false
+			if param.DataFormat == meta.ProfileDataFormatSVG {
+				svg, err := ConvertToSVG(data)
+				if err == nil {
+					data = svg
+					isSVG = true
+				}
+			}
+			if isSVG {
+				fileName += ".svg"
+			} else {
+				fileName += ".proto"
+			}
```

Comment:
I think you should throw an error. Otherwise when the other side asking ngm to provide a svg, ngm provides a proto format instead. It becomes ngm's fault.